src/server/saved_objects/service/lib/repository.js

```diff
@@ -606,4 +614,10 @@ export class SavedObjectsRepository {
     const savedObject = this._serializer.rawToSavedObject(raw);
     return omit(savedObject, 'namespace');
   }
+
+  assertAllowedType(type) {
+    if(this._schema.isHiddenType(type) && !this._includeHiddenTypes.includes(type)) {
+      throw Error('Hidden types not allowed!');
```

Comment:
@kobelb do you think we should throw a specific type of error here? We probably shouldn't say that the type is hidden, but maybe pretend like it's an unknown type?

Reply:
Agreed, throwing the same error for a hidden type and an unknown type would be best.

Reply:
Not that I'm aware of, we'll just want to ensure we're throw the same error for an unknown type and a hidden type.